astropy/utils/introspection.py

```diff
@@ -2,18 +2,17 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """Functions related to Python runtime introspection."""
 
-
 import inspect
-import re
 import os
 import sys
 import types
-import importlib
+import importlib.metadata
```

Comment:
`importlib.metadata` requires python 3.8, so this needs to be
```
try:
    from importlib import metadata
except ImportError:
    import importlib_metadata as metadata
```
(Unless we drop python 3.7, which would be borderline: it was released June 27, 2018, so we're supposed to support 42 months, i.e., until Dec 27, 2021...)

Reply:
Ah, I knew I missed something. Thanks!